### PR TITLE
Fixes after first implementation

### PR DESCRIPTION
**Pull request by rhouthuijzen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,21 +96,22 @@ The following settings are required to connect to the API.
 
 The `referenceId` contains the `externalId` of the person. This field is used within the _create_ lifecycle action to determine if an account exists and is part of the JSON payload to the target application.
 
-#### Error handling
-
-At this stage, the error handling functionality is still using the default function. This is because the error-handling logic could not be fully tested in all scenarios during the initial development phase.
+> [!IMPORTANT]
+> The referenceId can only be filled with the API or with a import (not in the GUI).
```
Comment (Rick-Jongbloed) [inline README.md]: a import -> an import

File: README.md [hunk 2]
```diff
@@ -96,21 +96,22 @@ The following settings are required to connect to the API.
...
-We recommend that error handling be revisited and thoroughly tested in various edge cases (e.g., invalid data formats, missing required fields, system failures, etc.) once more comprehensive tests can be conducted.
+#### Error handling
 
-For now, the default behavior will capture basic issues, but it may not provide customized or detailed feedback.
+Most of the errors in BasKMS are returned in the response. For this reason, the code will check the response if it contains an error.
 
 #### Social security number (BSN)
 
-The data returned by _KMS_ also contains the _social security number_ or _BSN_. Therefore, within the connector, both the output from `$correlatedAccount` and `$createdAccount` are filtered to only contain the fields specified in the field mapping or `$actionContext.Data` with the addition of the `id`.
+The data returned by _KMS_ also could contain the _social security number_ or _BSN_. Therefore, within the connector, both the output from `$correlatedAccount` and `$createdAccount` are filtered to only contain the fields specified in the field mapping or `$actionContext.Data` with the addition of the `id`.
```
Comment (Rick-Jongbloed) [inline README.md]: could also contain

File: README.md [hunk 3]
```diff
@@ -96,21 +96,22 @@ The following settings are required to connect to the API.
...
 
-#### UTF-8 encoding
+#### DepartmentName
+The fieldmapping contains a field `departmentName` this field is returned by BasKMS as `department.name` this is hardcoded to `departmentName` in the update script.
```
Comment (Rick-Jongbloed) [inline README.md]: Please use comma's and dots. 

The field mapping includes a field named departmentName, which is returned by BasKMS as department.name. In the update script, this is hardcoded to departmentName.

File: README.md [hunk 4]
```diff
@@ -96,21 +96,22 @@ The following settings are required to connect to the API.
...
 
-Before final deployment, comprehensive testing is required to validate that all data inputs and outputs are correctly encoded in UTF-8. This will include verifying that special characters and non-Latin scripts are accurately processed and displayed.
+> [!IMPORTANT]
+> If `departmentName` doesn't exist BasKMS will make the field  `department` empty. BasKMS won't return an error.
```
Comment (Rick-Jongbloed) [inline README.md]: If departmentName is not present, BasKMS will set the department field to empty without returning an error.

File: disable.ps1 [hunk 5]
```diff
@@ -101,16 +109,20 @@ try {
...
+                        id     = $actionContext.References.Account
                         active = $false
                     }
                 }
-                $null = Invoke-RestMethod @splatDisableUserParams
-            } else {
+                $response = Invoke-RestMethod @splatDisableUserParams
+                if ($response.error) {
+                    throw $($response.error)
+                }
```
Comment (Rick-Jongbloed) [inline disable.ps1:123]: Else? I think in the create and update script there's some more error handling?

File: disable.ps1 [hunk 6]
```diff
@@ -132,20 +144,22 @@ try {
...
+    else {
         $auditMessage = "Could not disable BasKMS account. Error: $($_.Exception.Message)"
         Write-Warning "Error at Line '$($ex.InvocationInfo.ScriptLineNumber)': $($ex.InvocationInfo.Line). Error: $($ex.Exception.Message)"
     }
     $outputContext.AuditLogs.Add([PSCustomObject]@{
-        Message = $auditMessage
-        IsError = $true
-    })
+            Message = $auditMessage
+            IsError = $true
```
Comment (Rick-Jongbloed) [inline disable.ps1:163]: could it be that an account doesn't exists? Would it then be better to skip the disable action (and set isError to false?)

File: fieldMapping.json [hunk 7]
```diff
@@ -97,28 +97,114 @@
...
+          "MappingMode": "Field",
+          "Value": "\"Person.PrimaryContract.Department.ExternalId\"",
+          "UsedInNotifications": false,
+          "StoreInAccountData": false
+        }
+      ]
+    },
+    {
+      "Name": "id",
+      "Description": "[required] account refference",
```
Comment (Rick-Jongbloed) [inline fieldMapping.json]: refference -> reference

**Review by Rick-Jongbloed — CHANGES_REQUESTED** [review]
Looks good, please have a look at the spelling and grammar issues en one question in the disable script

File: disable.ps1 (hunk 6 above)
Comment (rhouthuijzen) [reply disable.ps1:163]: Action NotFound solves this with IsError = $false
